diff --git a/share/functions/fish_vi_cursor.fish b/share/functions/fish_vi_cursor.fish
--- a/share/functions/fish_vi_cursor.fish
+++ b/share/functions/fish_vi_cursor.fish
@@ -35,7 +35,8 @@ function fish_vi_cursor -d 'Set cursor shape for different vi modes'
             and begin set -q KONSOLE_PROFILE_NAME
                 or set -q ITERM_PROFILE
                 or set -q VTE_VERSION # which version is already checked above
+                or string match -q "screen-256color" -- $TERM
                 or test (string replace -r "XTerm\((\d+)\)" '$1' -- $XTERM_VERSION) -ge 280
             end
             # .. unless an unsupporting terminal has been started in tmux inside a supporting one
