diff --git a/quirks/50-system-apple.quirks b/quirks/50-system-apple.quirks
--- a/quirks/50-system-apple.quirks
+++ b/quirks/50-system-apple.quirks
@@ -19,6 +19,16 @@ ModelAppleTouchpad=1
 MatchName=*Apple Inc. Apple Internal Keyboard*
 AttrKeyboardIntegration=internal
 
+[Apple SPI Touchpad]
+MatchName=*Apple SPI Touchpad*
+ModelAppleTouchpad=1
+AttrTouchSizeRange=50:30
+AttrPalmSizeThreshold=800
+
+[Apple SPI Keyboard]
+MatchName=*Apple SPI Keyboard*
+AttrKeyboardIntegration=internal
+
 [Apple MagicMouse]
 MatchUdevType=mouse
 MatchBus=bluetooth
