From 093ae6e2e32a3cbbc6c046df0085d9ca07a602cb Mon Sep 17 00:00:00 2001
From: Marta Lofstedt <marta.lofstedt@intel.com>
Date: Wed, 2 Nov 2016 09:33:24 +0200
Subject: backlight: expose the current brightness in the new kernel API

Signed-off-by: Marta Lofstedt <marta.lofstedt@intel.com>

diff --git a/drivers/video/backlight/backlight.c b/drivers/video/backlight/backlight.c
index 8af860f..04e28d5 100644
--- a/drivers/video/backlight/backlight.c
+++ b/drivers/video/backlight/backlight.c
@@ -188,6 +188,12 @@ int backlight_device_set_brightness(struct backlight_device *bd,
 }
 EXPORT_SYMBOL(backlight_device_set_brightness);
 
+int backlight_device_get_brightness(struct backlight_device *bd)
+{
+	return bd->props.brightness;
+}
+EXPORT_SYMBOL(backlight_device_get_brightness);
+
 static ssize_t brightness_store(struct device *dev,
 		struct device_attribute *attr, const char *buf, size_t count)
 {
diff --git a/include/linux/backlight.h b/include/linux/backlight.h
index 895c4661..5adb2ea 100644
--- a/include/linux/backlight.h
+++ b/include/linux/backlight.h
@@ -145,6 +145,7 @@ extern int backlight_register_notifier(struct notifier_block *nb);
 extern int backlight_unregister_notifier(struct notifier_block *nb);
 extern struct backlight_device *backlight_device_get_by_type(enum backlight_type type);
 extern int backlight_device_set_brightness(struct backlight_device *bd, unsigned long brightness);
+extern int backlight_device_get_brightness(struct backlight_device *bd);
 
 struct backlight_device *backlight_device_lookup(const char *name);
 void backlight_set_brightness(struct backlight_device *bd, unsigned int value,
-- 
cgit v0.10.2

