From 26c20754e7adbabc1d15cc566b127861cabde86b Mon Sep 17 00:00:00 2001
From: Marta Lofstedt <marta.lofstedt@intel.com>
Date: Thu, 13 Oct 2016 14:34:18 +0300
Subject: i915: use drm backlight

Allocate and set the name with drm backlight.

Signed-off-by: Marta Lofstedt <marta.lofstedt@intel.com>

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 1a6ff26..02c08ab 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -34,6 +34,7 @@
 #include <linux/moduleparam.h>
 #include <linux/pwm.h>
 #include "intel_drv.h"
+#include <drm/drm_backlight.h>
 
 #define CRC_PMIC_PWM_PERIOD_NS	21333
 
@@ -1706,6 +1707,10 @@ int intel_panel_setup_backlight(struct drm_connector *connector, enum pipe pipe)
 
 	panel->backlight.present = true;
 
+	/* This connector has backlight attached */
+	if (!drm_backlight_alloc(connector))
+		drm_backlight_set_name(connector->backlight, "intel_backlight");
+
 	DRM_DEBUG_KMS("Connector %s backlight initialized, %s, brightness %u/%u\n",
 		      connector->name,
 		      enableddisabled(panel->backlight.enabled),
-- 
cgit v0.10.2

